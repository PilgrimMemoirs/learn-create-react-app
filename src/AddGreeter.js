import React, { Component } from 'react';
import './AddGreeter.css';


export default AddGreeter;
