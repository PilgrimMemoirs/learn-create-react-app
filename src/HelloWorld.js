import React, { Component } from 'react';
import './App.css';

class HelloWorld extends Component {
  constructor(props){
    super(props);
    this.state = {greeting: 'Hello'};
    this.frenchify = this.frenchify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }//constructor enabled
  removeGreeting(){
    this.props.removeGreeting(this.props.name);
  }
  render () {
    return (
      <div className="HelloWorld">
        <h2>{this.state.greeting} {this.props.name}!</h2>
        <button onClick={this.frenchify}> Frenchify! </button>
        <br>
        <button onClick={this.removeGreeting}>Remove Me!</button>
      </div>
    );
  } //render end
  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }

};//end HelloWorld class

export default HelloWorld;
