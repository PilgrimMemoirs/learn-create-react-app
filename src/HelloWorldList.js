import React, { Component } from 'react';
import './HelloWorldList.css';

import HelloWorld from './HelloWorld';


class HelloWorldList extends Component {
  render() {
    return (
      <div className="HelloWorldList">
        <HelloWorld name="Amper"/>
        <HelloWorld name="Octo"/>
      </div>
    )
  }
}


export default HelloWorldList;
