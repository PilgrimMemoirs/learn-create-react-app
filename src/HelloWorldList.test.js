import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import AddGreeter from './AddGreeter';
import HelloWorld from './HelloWorld';

describe (HelloWorldList, () => {
  const component = shallow(
    <HelloWorldList />
  );
});
