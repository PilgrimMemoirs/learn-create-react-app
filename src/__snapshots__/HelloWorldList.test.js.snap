// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorldList extends _react.Component {
  constructor(props) {
    super(props);
    this.state = { greetings: ['Amper', 'Octo', 'Jamie'] };
    this.addGreeting = this.addGreeting.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  addGreeting(newName) {
    this.setState({ greetings: [...this.state.greetings, newName] });
  }
  renderGreetings() {
    return this.state.greetings.map(name =>
    _react2.default.createElement(_HelloWorld2.default, {
      key: name,
      name: name,
      removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this }));


  }
  removeGreeting(removeName) {
    const filteredGreetings = this.state.greetings.filter(name => {
      return name !== removeName;
    });
    this.setState({ greetings: filteredGreetings });
  }
  render() {
    return (
      _react2.default.createElement('div', { className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 35 }, __self: this },
        _react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 36 }, __self: this }),
        this.renderGreetings()));


  }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddGreeter"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    <h2>
      Hello
       
      Amper
      !
    </h2>
    <button
      className="frenchify"
      onClick={[Function]}
    >
       Frenchify! 
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    <h2>
      Hello
       
      Octo
      !
    </h2>
    <button
      className="frenchify"
      onClick={[Function]}
    >
       Frenchify! 
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    <h2>
      Hello
       
      Jamie
      !
    </h2>
    <button
      className="frenchify"
      onClick={[Function]}
    >
       Frenchify! 
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
</div>
`;
